packages/devextreme-react/gulpfile.js

```diff
@@ -111,5 +156,73 @@ gulp.task(NPM_BUILD_WITH_HEADERS, gulp.series(
 
 gulp.task(NPM_PACK, gulp.series(
     NPM_BUILD_WITH_HEADERS,
+    gulp.parallel(
+        NPM_PREPARE_MODULES,
+        NPM_PREPARE_FOLDERS,
+    ),
     shell.task(['npm pack'], {cwd: config.npm.dist})
 ));
+
+function makeModuleForFolder(folderPath, moduleFileNames) {
+    const distFolder = path.join(__dirname, config.npm.dist);
+
+    moduleFileNames = moduleFileNames || findJsModuleFileNamesInFolder(path.join(distFolder, 'esm', folderPath));
+
+    try {
+        fs.mkdirSync(path.join(distFolder, folderPath));
+
+        if (fs.existsSync(path.join(distFolder, 'esm', folderPath, 'index.js'))) {
+            generatePackageJsonFile(folderPath);
+        }
+
+        moduleFileNames.forEach((moduleName) => {
+            makeModuleForFile(moduleName, folderPath);
+        })
+    } catch (error) {
+        error.message = `Exception while makeModuleForFolder(${folderPath}).\n ${error.message}`;
+        throw(error);
+    }
+}
+
+function makeModuleForFile(moduleName, folder = null) {
+    fs.mkdirSync(path.join(__dirname, config.npm.dist, folder || '', moduleName));
+
+    generatePackageJsonFile(folder, moduleName);
+}
+
+function generatePackageJsonFile(folder, moduleName) {
+    moduleName = moduleName || '';
+    const absoluteModulePath = path.join(__dirname, config.npm.dist, folder || '', moduleName);
+    const moduleFilePath = (folder ? folder + '/' : '') + (moduleName || 'index');
```

Comment:
Use path.join to create path. Also, seems that you can move it to [199 line](https://github.com/DevExpress/DevExtreme/pull/25483/files#diff-ba5fd6cb7491c97328544cdc73f6d49547ff0ecc3ae4d2b06da2dd2cd19b3c43R199)

Reply:
path.join uses OS specific delimiter but we need only '/' for package.json